File: go/parity_test.go
package likeness

import (
	"fmt"
	"os"
	"path/filepath"
	"testing"
)

func TestEmitParity(t *testing.T) {
	out := os.Getenv("LIKENESS_PARITY_OUT")
	if "" == out {
		t.Skip("LIKENESS_PARITY_OUT is not set")
	}
	root := Root(t)
	for _, e := range corpusEntries(t, root, "cli") {
		r := Run(buildCtx(t, root, e))
		dst := filepath.Join(out, Port, str(e, "id")+".out")
		if err := os.MkdirAll(filepath.Dir(dst), 0o755); nil != err {
			t.Fatal(err)
		}
		body := fmt.Sprintf("exit %d\n--stdout--\n%s--stderr--\n%s", r.Exit, r.Stdout, r.Stderr)
		if err := os.WriteFile(dst, []byte(body), 0o644); nil != err {
			t.Fatal(err)
		}
	}
}

File: typescript/test/parity.test.ts

import { test } from 'node:test'
import { mkdirSync, writeFileSync } from 'node:fs'
import { dirname, join } from 'node:path'

import { run, PORT } from '../src/core.js'
import { buildCtx, entries, type CliEntry } from './harness.js'

const OUT = process.env.LIKENESS_PARITY_OUT

test('emit parity outputs', { skip: undefined === OUT ? 'LIKENESS_PARITY_OUT is not set' : false },
  async () => {
    const base = join(OUT as string, PORT)
    for (const e of entries<CliEntry>('cli')) {
      const r = await run(buildCtx(e))
      const dst = join(base, e.id + '.out')
      mkdirSync(dirname(dst), { recursive: true })
      // exit, stdout and stderr in one file, so a difference in any of the
      // three is one comparison rather than three.
      writeFileSync(dst, 'exit ' + r.exit + '\n--stdout--\n' + r.stdout + '--stderr--\n' + r.stderr)
    }
  })
